### PR TITLE
Fixes "Burning rust off walls doesn't blind you"

### LINKED ISSUE #68796: Burning rust off walls doesn't blind you
## Round ID:
Tested locally.

## Reproduction:
Go to burn rust off a wall, your eyes are fine.
Go do ANYTHING else with a welder, your eyes are NOT fine.

### PR DESCRIPTION
**Pull request by CapybaraExtravagante**
## About The Pull Request

This fixes the bug where burning rust off of a wall using a welder would not actually blind you! It was happening because it was using the wrong procs to use the tools. It now correctly implements tool_start_check and tool_start_check

fixes: #68796

## Why It's Good For The Game

Less bugs = good

## Changelog

:cl: Capybara Holly
fix: Trying to remove rust from walls with a welding tool will now blind you like it should
/:cl:

### CODE REVIEW

File: code/datums/elements/rust.dm [hunk 1]
```diff
@@ -47,16 +47,23 @@
...
+			Detach(source)
+			return
+
+
 		if(TOOL_RUSTSCRAPER)
+			if(!item.tool_start_check(user, amount=0))
+				return
 			user.balloon_alert(user, "scraping off rust...")
-			if(!do_after(user, 2 SECONDS * item.toolspeed, source))
+			if(!item.use_tool(source, user, 2 SECONDS * item.toolspeed))
```
Comment (Fikou) [inline code/datums/elements/rust.dm:66]: use_tool will already include toolspeed

File: code/datums/elements/rust.dm [hunk 2]
```diff
@@ -47,16 +47,23 @@
...
-				if(!do_after(user, 5 SECONDS * item.toolspeed, source))
-					return
-				user.balloon_alert(user, "burned off rust")
-				Detach(source)
+			if(!item.tool_start_check(user, amount=5))
 				return
+
+			user.balloon_alert(user, "burning off rust...")
+
+			if(!item.use_tool(source, user, 5))
```
Comment (Fikou) [inline code/datums/elements/rust.dm:55]: is 5 time here? used named arguments and/or seconds defines

File: code/datums/elements/rust.dm [hunk 3]
```diff
@@ -47,16 +47,23 @@
...
+				return
+			user.balloon_alert(user, "burned off rust")
+			Detach(source)
+			return
+
+
 		if(TOOL_RUSTSCRAPER)
+			if(!item.tool_start_check(user, amount=0))
```
Comment (Fikou) [inline code/datums/elements/rust.dm:63]: no need for amount here i think

**Comment by optimumtact** [comment]
dupe of #68945 unfortunately

@SuperNovaa41 which one is more correct?

**Comment by SuperNovaa41** [comment]
> dupe of #68945 unfortunately
> 
> @SuperNovaa41 which one is more correct?

mine was lazy this is better

**Comment by Jolly-66** [comment]
Don't forget to put `fixes: 68796` in your PR body to gave the issue linked and close when the PR is merged

**Pull request merged**